### PR TITLE
To Do list: interactive list

### PR DESCRIPTION
**Pull request by TadesseDev**
## Changes introduced ✍🏼
---
- Use modules in a professional way. 😎
- Add event listener for the checkbox to mark tasks as completed or reversed. ✔
- Update object data on checkbox event. ✔
- Create an event listener to clear all completed actions. 👌
- Store data and status in local storage. 🎞
- preserve checkbox state on reload (inside the DOM). 🙌
- 🔏 Linter is not working on git hub for a moment (but i have tested it locally)

### CODE REVIEW

File: src/modules/MyTasks.js [hunk 1]
```diff
@@ -60,8 +83,15 @@ class MyTasks {
     section.insertBefore(getTaskItem(this), clear);
   }
 
+  removeFromDom = () => {
+    const element = document.getElementById(String(this.index));
+    element.remove();
+  }
+
```
Comment (GzimAsani) [inline src/modules/MyTasks.js]: - [ ] It is required that "Clear all completed" button should use the `Array.filter()` method.

File: src/modules/methods.js [hunk 2]
```diff
@@ -34,19 +43,37 @@ const alterTask = (element, deleteIcon, task) => {
   });
```
Comment (GzimAsani) [inline src/modules/methods.js:42]: - [ ] Please remove all commented-out code that you do not intend to use. Such as the one on line 29 on the methods.js file.

**Review by GzimAsani — CHANGES_REQUESTED** [review]
# Hi  @Tadesse-Alemayehu 
Good job so far! You are almost done :+1:  :+1: **_Your linters are working, but for some reason they are not showing on github_**
There are some issues that you still need to work on to go to the next project but you are almost there!

## To highlight:
- Good job with the gitflow :+1:
- Good PR description :clap: 

## Required Changes ♻️ 
- [ ] _Check the comments below :arrow_double_down: _

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/modules/MyTasks.js (hunk 1 above)
Comment (TadesseDev) [reply src/modules/MyTasks.js]: @GzimAsani @thank you for the review. I'll make the changes right away.

File: src/modules/methods.js (hunk 2 above)
Comment (TadesseDev) [reply src/modules/methods.js:42]: Dear @GzimAsani I have made the changes. I have used  filter Method at https://github.com/Tadesse-Alemayehu/To-Do-list/pull/7/commits/a9e93caaa2c8622ff649b2c19762a4fa9fca0b86 commit hash and remove comments in https://github.com/Tadesse-Alemayehu/To-Do-list/pull/7/commits/80edb5cfd635086a2185e1703670d7c1ac9994f2 commit hash. Thank you. 😎

**Review by Lordkaito — APPROVED** [review]
Hi @Tadesse-Alemayehu, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**